pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -1374,6 +1374,12 @@ func setInstanceMetadataHostname(data map[string]interface{}) {
 				// set default false for aws cloud provider
 				values.PutValue(data, false, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
 			}
+		} else if cloudProviderName == k8s.ExternalAWSCloudProviderName {
+			_, ok := values.GetValue(data, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
+			if !ok {
+				// set default true for external aws cloud provider
+				values.PutValue(data, true, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
+			}
```

Comment:
We can simplify the blob `if ok && rkeConfig != nil {`:
``` go
	if ok && rkeConfig != nil {
		cloudProviderName := convert.ToString(values.GetValueN(data, "rancherKubernetesEngineConfig", "cloudProvider", "name"))
		_, ok := values.GetValue(data, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
		if !ok {
			var val bool
			switch cloudProviderName {
			case k8s.AWSCloudProvider:
				// set default false for aws cloud provider
				val = false
			case k8s.ExternalAWSCloudProviderName:
				// set default true for external aws cloud provider
				val = true
			}
			values.PutValue(data, val, "rancherKubernetesEngineConfig", "cloudProvider", "useInstanceMetadataHostname")
		}
	}
```

Reply:
cleaned it up